src/test/lib/optimizer/strategy/predicate_placement_rule_test.cpp

```diff
@@ -247,59 +254,76 @@ TEST_F(PredicatePlacementRuleTest, ComplexBlockingPredicatesPushdownTest) {
 }
 
 TEST_F(PredicatePlacementRuleTest, AllowedValuePredicatePushdownThroughProjectionTest) {
-  // We can push `a > 4` under the projection because it does not depend on the subquery.
+  // We can push `a > 4` under the projection because it does not depend on the sub-query.
+  // clang-format off
+  const auto input_lqp =
+  PredicateNode::make(greater_than_(_a_a, value_(4)),
+    ProjectionNode::make(expression_vector(_a_a, _a_b, _subquery_c),
+      _table_a));
 
-  auto predicate_node = std::make_shared<PredicateNode>(greater_than_(_a_a, value_(4)));
-  predicate_node->set_left_input(_projection_pushdown_node);
+  const auto expected_lqp =
+  ProjectionNode::make(expression_vector(_a_a, _a_b, _subquery_c),
+    PredicateNode::make(greater_than_(_a_a, value_(4)),
+       _table_a));
+  // clang-format on
 
-  auto reordered = StrategyBaseTest::apply_rule(_rule, predicate_node);
+  auto actual_lqp = StrategyBaseTest::apply_rule(_rule, input_lqp);
 
-  EXPECT_EQ(reordered, _projection_pushdown_node);
-  EXPECT_EQ(reordered->left_input(), predicate_node);
-  EXPECT_EQ(reordered->left_input()->left_input(), _table_a);
+  // TODO(Marcel) If a node mapping is created in `lqp_find_subplan_mismatch` which is called in `EXPECT_LQP_EQ`, the
+  // nodes of sub-queries are not mapped. Consequently, `expression_adapt_to_different_lqp` cannot find the
+  // StoredTableNode of the sub-query in the node mapping. Therefore, an assertion fails: "Couldn't find referenced
+  // node x in NodeMapping."
+  EXPECT_LQP_EQ(actual_lqp, expected_lqp);
```

Comment:
This is the issue that I tried to explain during today's Fortnightly. If anyone had this issue before and knows a good way to solve it, it would be a pleasure to get feedback.